modin/distributed/dataframe/pandas/partitions.py

```diff
@@ -15,6 +15,12 @@
 
 import numpy as np
 
+from modin.core.execution.ray.implementations.pandas_on_ray.partitioning import (
+    PandasOnRayDataframePartition,
+)
+from modin.core.execution.dask.implementations.pandas_on_dask.partitioning.partition import (
+    PandasOnDaskDataframePartition,
+)
```

Comment:
this feels wrong, it would make _both_ `ray` and `dask` hard requirements installed before running things regardless of which engine we plan to use

Reply:
Reverted.